src/store/modules/product/actions.ts

```diff
@@ -75,6 +92,40 @@ const actions: ActionTree<ProductState, RootState> = {
     return resp;
   },
 
+  async getVariants({ dispatch, commit }, payload) {
+    if (payload.viewIndex === 0) emitter.emit("presentLoader");
+    
+    // checking if product and its variants are in cache
+    let product = JSON.parse(JSON.stringify(this.getters['product/getProduct'](payload.productId)))
```

Comment:
Instead of using getters, use state.

Reply:
Used fetchProduct action's response as discussed

Reply:
Updated the code as per discussion.